dmv/dmv.js

```diff
@@ -19,13 +19,18 @@ DMV.prototype.currentCustomerFor = function(agent) {
 }
 
 DMV.prototype.nextCustomer = function() {
+  if (this.customers.length === 0) {
+    throw new Error('No customers')
+  }
   const currentCustomer = this.customers.shift()
-  this.agentList.find(agent => {
+  let currentAgent = null;
+  this.agentList.forEach(agent => {
     if (!this.attending[agent]) {
       this.attending[agent] = currentCustomer
-      return true
+      currentAgent = agent
     }
   })
+  return {agent: currentAgent, customer: currentCustomer}
 }
```

Comment:
I this function encapsulates a lot of items.  I think that the method with in it is written cleanly and easy to follow which helps a lot with the readability. I also appreciate that the error catch at the top of the prototype but I think the code could be reordered to help further readability.